### PR TITLE
DOC: updated docstring for deprecation of axis=1 in groupby

### LINKED ISSUE #54873: DEPR: Add note to groupby docstring about deprecation of axis=1
Ref: https://github.com/pandas-dev/pandas/pull/51395#issuecomment-1699708761

The docstring of both DataFrame.groupby and Series.groupby should be updated to indicate the deprecation. Marking as a regression to make clear this goes into 2.1.1.

### PR DESCRIPTION
**Pull request by rsm-23**
- [x] closes #54873
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

**Comment by rsm-23** [comment]
@rhshadrach @mroeschke I added a note on the deprecation. Would be great if you could give some inputs here?

File: pandas/core/shared_docs.py [hunk 1]
```diff
@@ -113,6 +113,13 @@
 axis : {0 or 'index', 1 or 'columns'}, default 0
     Split along rows (0) or columns (1). For `Series` this parameter
     is unused and defaults to 0.
+
+    .. deprecated:: 2.1.0
+
+        This now defaults to 0 for both `DataFrame` and `Series`.
```
Comment (mroeschke) [inline pandas/core/shared_docs.py:119]:
```suggestion
        Will be removed and behave like axis=0 in a future version.
```

File: pandas/core/shared_docs.py [hunk 2]
```diff
@@ -113,6 +113,13 @@
...
+        DataFrame.groupby with axis=1 is deprecated. Do
+        Instead, do `frame.T.groupby(...)` without axis.
```
Comment (mroeschke) [inline pandas/core/shared_docs.py:121]:
```suggestion
        For ``axis=1``, do ``frame.T.groupby(...)`` instead.
```

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Pull request merged**